pkg/ingester/index/index.go

```diff
@@ -163,13 +181,44 @@ func (ii *invertedIndex) delete(labels labelPairs, fp model.Fingerprint) {
 		}
 
 		if len(values) == 0 {
-			delete(i.idx, name)
+			delete(shard.idx, name)
 		} else {
-			i.idx[name] = values
+			shard.idx[name] = values
 		}
 	}
 }
 
+func (shard *indexShard) labelNames() model.LabelNames {
+	shard.mtx.RLock()
+	defer shard.mtx.RUnlock()
+
+	results := make(model.LabelNames, 0, len(shard.idx))
+	for name := range shard.idx {
+		results = append(results, name)
+	}
+
+	sort.Sort(labelNames(results))
+	return results
+}
+
+func (shard *indexShard) labelValues(name model.LabelName) model.LabelValues {
```

Comment:
It looks like if you swap the order of `delete()` and `labelValues()` in the file, the diff will come out smaller and easier to read.

Reply:
I've reordered them but the diff isn't any smaller; sorry.